src/data/dfinsupp/basic.lean

```diff
@@ -1040,6 +1044,121 @@ assume f g, decidable_of_iff (f.support = g.support ∧ (∀i∈f.support, f i =
+variables {α : ι → Type*} {δ : (Σ i, α i) → Type v} [Π i, has_zero (δ i)]
+
+/--The natural map between `Π₀ (i : Σ i, α i), δ i` and `Π₀ i (j : α i), δ ⟨i, j⟩`.-/
+noncomputable def curry (f : Π₀ i, δ i) : Π₀ i j, δ ⟨i, j⟩ := by { classical, exact
+mk (f.support.image $ λ i, i.1)
+  (λ i, mk (f.support.preimage (sigma.mk i) $ sigma_mk_injective.inj_on _) $ λ j, f ⟨i, j⟩) }
+
+@[simp] lemma curry_apply (f : Π₀ i, δ i) (i : ι) (j : α i) : curry f i j = f ⟨i, j⟩ :=
+begin
+  dunfold curry, by_cases h : f ⟨i, j⟩ = 0,
+  { rw [h, mk_apply], split_ifs, { rw mk_apply, split_ifs, { exact h }, { refl } }, { refl } },
+  { rw [mk_of_mem, mk_of_mem], { refl },
+    { rw [mem_preimage, mem_support_to_fun], exact h },
+    { rw mem_image, refine ⟨⟨i, j⟩, _, rfl⟩, rw mem_support_to_fun, exact h } }
+end
+
+/--The natural map between `Π₀ i (j : α i), δ ⟨i, j⟩` and `Π₀ (i : Σ i, α i), δ i`, inverse of
+`curry`.-/
+noncomputable def uncurry (f : Π₀ i j, δ ⟨i, j⟩) : Π₀ i, δ i := by { classical, exact
+mk (f.support.bUnion $ λ i, (f i).support.image $ sigma.mk i) (λ ⟨⟨i, j⟩, _⟩, f i j) }
+
+@[simp] lemma uncurry_apply (f : Π₀ i j, δ ⟨i, j⟩) (i : ι) (j : α i) : uncurry f ⟨i, j⟩ = f i j :=
+begin
+  dunfold uncurry, by_cases h : f i j = 0,
+  { rw mk_apply, split_ifs, { refl }, { exact h.symm } },
+  { apply mk_of_mem, rw mem_bUnion, refine ⟨i, _, _⟩,
+    { rw mem_support_to_fun, intro H, rw ext_iff at H, exact h (H j) },
+    { apply mem_image_of_mem, rw mem_support_to_fun, exact h } }
+end
+
+/--The natural bijection between `Π₀ (i : Σ i, α i), δ i` and `Π₀ i (j : α i), δ ⟨i, j⟩`.-/
+noncomputable def curry_equiv : (Π₀ i, δ i) ≃ Π₀ i j, δ ⟨i, j⟩ := ⟨curry, uncurry,
+λ f, by { ext ⟨i, j⟩, rw [uncurry_apply, curry_apply] },
+λ f, by { ext i j, rw [curry_apply, uncurry_apply] }⟩
+
+end curry
+
+variables {α : option ι → Type v} [Π i, has_zero (α i)] --[Π i (x : α i), decidable (x ≠ 0)]
+
+/--Adds a term to a dfinsupp, making a dfinsupp indexed by an `option`.-/
+def extend_with (a : α none) : (Π₀ i, α (some i)) → Π₀ i, α i :=
+begin
+  refine quotient.lift (λ f, ⟦_⟧) (λ f f' h, _),
+  exact { to_fun := option.rec a f.to_fun,
+    pre_support := none ::ₘ (f.pre_support.map some),
+    zero := λ i, option.rec (or.inl $ multiset.mem_cons_self _ _)
+      (λ i, (f.zero i).imp_left $ λ h, multiset.mem_cons_of_mem $ multiset.mem_map_of_mem _ h) i },
+  { refine quot.sound (option.rec _ $ λ x, _), refl, exact h x },
+end
+@[simp] lemma extend_with_none (f : Π₀ i, α (some i)) (a : α none) : f.extend_with a none = a :=
+by { rcases f, refl }
+@[simp] lemma extend_with_some (f : Π₀ i, α (some i)) (a : α none) (i : ι) :
+f.extend_with a (some i) = f i :=
+by { rcases f, refl }
+
+include dec
+/--Bijection obtained by separating the term of index `none` of a dfinsupp over `option ι`.-/
+@[simps] noncomputable def equiv_prod_dfinsupp : (Π₀ i, α i) ≃ α none × Π₀ i, α (some i) :=
+⟨λ f, (f none, comap_domain some (option.some_injective _) f), λ f, f.2.extend_with f.1,
```

Comment:
Can't you use the computable version here? The inverse function is `get_or_else 0`, isn't it?

Reply:
You're right, this was a mistake on my parts